opm/material/fluidmatrixinteractions/EclMaterialLawManager.hpp

```diff
@@ -346,8 +358,76 @@ class EclMaterialLawManager
+                               gasOilUnscaledPointsVector, oilWaterUnscaledPointsVector, gasOilEffectiveParamVector,oilWaterEffectiveParamVector,
+                               epsGridProperties, epsImbGridProperties );
+
 
+        // for well connections
+        std::vector<int> satnumRegionArrayConnections;
+        std::vector<int> compressedToCartesianElemIdxConnections;
+
+        const auto& schedule = eclState.getSchedule();
+        const auto& eclGrid = eclState.getInputGrid();
+
+        size_t timeStep = 0;
+        auto wells = schedule.getWells(timeStep); //need all wells ...
+        for (auto wellIter= wells.begin(); wellIter != wells.end(); ++wellIter) {
+            const auto* well = (*wellIter);
+            for(const auto& completion : well->getCompletions(timeStep)) { // ... and all connections!!!
+
+                int cartIdx = eclGrid.getGlobalIndex(completion.getI(), completion.getJ(), completion.getK());
+                compressedToCartesianElemIdxConnections.push_back(cartIdx);
+
+                int satnumId = completion.getSatTableId();
+                if (satnumId > 0)
+                    satnumRegionArrayConnections.push_back(satnumId);
+                else {
+                    int satnumIdFromCell = satnumRegionArray[eclGrid.activeIndex(cartIdx)];
+                    satnumRegionArrayConnections.push_back(satnumIdFromCell);
+                }
+            }
         }
+        // TODO change satnum/imbnum in epsGridProperties or change in ->extractScaled(...) ?
+        EclEpsGridProperties epsConnectionsProperties = epsGridProperties;
+        EclEpsGridProperties epsImbConnectionsProperties = epsImbGridProperties;
+
+        initParamsForElements_(materialLawParamsForWellConnections_, deck, eclState, compressedToCartesianElemIdxConnections, satnumRegionArrayConnections,
+                               gasOilUnscaledPointsVector, oilWaterUnscaledPointsVector, gasOilEffectiveParamVector,oilWaterEffectiveParamVector,
+                               epsConnectionsProperties, epsImbConnectionsProperties );
+
+
+
+    }
+
+
+    void initParamsForElements_(std::vector<std::shared_ptr<MaterialLawParams> > materialLawParams,
+                                const Deck& deck, const EclipseState& eclState,
+                                const std::vector<int>& compressedToCartesianElemIdx,
+                                const std::vector<int>& satnumRegionArray,
+                                const GasOilScalingPointsVector& gasOilUnscaledPointsVector,
+                                const OilWaterScalingPointsVector& oilWaterUnscaledPointsVector,
+                                const GasOilEffectiveParamVector& gasOilEffectiveParamVector,
+                                const OilWaterEffectiveParamVector& oilWaterEffectiveParamVector,
+                                const EclEpsGridProperties& epsGridProperties,
+                                const EclEpsGridProperties& epsImbGridProperties)
```

Comment:
I don't really like the signature of this method: Remember the "rule of seven"? since it is an internal method, I can live with it if it needs to be, though...